### ISSUE TITLE
The button "´" used to create "É", "é", "ć" and so on, does not work

### ISSUE THREAD

**Issue by Jacalz** [fyne-io/fyne#590]
**Describe the bug**
When typing in an entry widget, the button with the symbol **´** does not work and makes it impossible to type letters like "É", "é", "ć" and so on. You are supposed to first press **´** and then a letter and you should then get something like É or ć depending on what letter you pressed after. See https://en.wikipedia.org/wiki/%C3%89 for more information on the usage in various languages.

Copy pasting the letter makes it show correctly in the entry widget. Might be that the key just isn't checked for presses or something along those lines.

**To Reproduce**
1. Focus an entry widget of your liking
2. Click on the **´** button on your keyboard.
3. Press any letter of your liking for example **e**, **E** or **a**.
4. Look and see that nothing happens. Not even the letter is shown.

**Device (please complete the following information):**
 - OS: Solus (A Linux Distrubution)
 - Version: Solus 4.0 using Linux Kernel 5.3.18
 - Go version: `1.13.5`
 - Fyne version: 1.1.2, 1.2.0 have all been tested and does show this bug.

**Comment by andydotxyz** [comment]
If one does not have a "´" button on their keyboard how can we do this?

**Comment by andydotxyz** [comment]
Is this possibly a duplicate of the previous issue discussing compose keys? #38

**Comment by ohir** [comment]
@andydotxyz 
> Is this possibly a duplicate of the previous issue discussing compose keys? #38

Not exact duplicate, but one of #38 reported bugs is still here.
I can confirm that fyne somewhat managed to get into the compose ways, ie on my system it just is eating dead keys.

> If one does not have a "´" button on their keyboard how can we do this?

Dead keys are mapped to different keys, or chords with different keyboard layouts.
 - US MacOS: go to keyboard settings and check status visible, then open the layout helper from under the keyboard layouts' topbar icon. Dead keys (with opt) are in orange. 
 - US Windows: install second US layout known as US-Intl, this also has dead keys exposed. [See more](https://docs.microsoft.com/en-us/globalization/windows-keyboard-layouts#) at technet site.
- Linux: see manuals: xmodmap, setxkbmap, xkeyboard-config, and xev. You can use [us-ansi based linux layout](https://play.golang.org/p/wl-LrZ-vaBt) with many dead accents mapped to Menu-Fx I made for US developers.

PS. with a stock GB layout the acute dead accent is located at AltGr+'@ key. See [red captions here](https://en.wikipedia.org/wiki/File:KB_United_Kingdom_Ext.png).

**Comment by Jacalz** [comment]
I think you have one @andydotxyz, it is (on Swedish keyboard layouts) the same key you use to make special text in markdown with `like how this text looks`. The same key, just that you don't press shift and then press the letter of your choice.

From looking at pictures of brittish keyboard layouts, it looks like it might be the key to the left of your number one (1 and !) key under the escape key.

**Comment by andydotxyz** [comment]
When I test locally, it seems to work, Alt+` then a gets à printed in the entry.
Is this not correct and/or not working on your setup?

**Comment by ohir** [comment]
> Alt+` then a gets à printed in the entry.
> Is this not correct and/or not working on your setup?

This is a **direct** production (a Lvl3_ISO chord) for the a with **grave**. This, ie. direct production works fine.

It is the dead-key compose method that does not work. While some national layouts use only chords, many do not - there dead keys are used to enter "everyday" characters. With dead key you first press the dead accent **then** you press another key. So on GB-intl you press AltGr+' (dead acute - nothing prints yet), then **a**. And only after that **á** should appear. (Note the skew direction).

**Comment by toaster** [comment]
Hi, I'm running MacOS 10.14 with a German keyboard layout and it works as expected (pressing `´`, nothing shows, pressing `e`, `é` shows).
However, it differs from normal OS behaviour in the way that the press of `´` does not show anything until pressed again. Normally the `´` is shown underscored to indicate it as pending. If a key suitable for combination is pressed then, the `´` is replaced by the combined character. If another key is hit, the `´` becomes normal (not underlined) and the character of the other key is printed, too.

**Comment by Jacalz** [comment]
I might need to test this on MacOS or Windows to see if I can replicate it there. Don't work for me on Linux anyhow.

**Comment by andydotxyz** [comment]
It works on my linux as well. Tilo is right it could probably be more polished, but I don't know how given the GLFW driver - we could look into it more.
Perhaps it relates to the definition of dead keys rather than the functionality itself?

**Comment by Jacalz** [comment]
Yeah it might be something along those lines and maybe related to keyboard layouts depending on language.

**Comment by toaster** [comment]
I'll check on Linux/Gnome with German language/keyboard settings as soon as I set my system up (Gentoo on a 13 years old box … might take a while to compile everything :)).

**Comment by Jacalz** [comment]
I have done some more testing and I seem to get this to work both on MacOS and Windows, however I just can't seem to get it working on Linux. I don't know if it handles keyboard presses differently.

**Comment by andydotxyz** [comment]
Hmm, strange on going back to my linux computer today I see that AltGr+;, AltGr+' and AltGr+# seem to be the main ones with AltGr+{ and AltGr+} augmenting letters as well, different to macOS but nonetheless working...

**Comment by Jacalz** [comment]
Sorry for the late answer, but it has never been about pressing AltGr for me, but shift is the modifier on the Swedish layout. This is really strange indeed...

**Comment by andydotxyz** [comment]
I'm not too sure what we can do to progress this sorry.
Until someone can outline how it should work on linux, and how we're not meeting that system, it's a bit challenging. Let me know if @Jacalz or @toaster manage to configure and figure this out :).

**Comment by toaster** [comment]
I did some testing on Linux (Gentoo, Gnome, German keyboard) and found the following:
GLFW does not call the character callback when entering combined characters using a dead key (like `´`) and a normal key (`´`+ `e` -> `é`). The key callback gets called. I assume there is a bug in the GLFW input handling logic. This should be checked with the C library (without Go binding) before reporting it upstream.

**Comment by andydotxyz** [comment]
Good catch. If you are making changes to the C code and rebuilding be sure to clear the caches in between builds - I got caught by this before where the glfw source was not actually rebuilding!

**Comment by LaPingvino** [comment]
I encounter this issue when compiling on ARM64 Debian in a VM under ChromeOS (Lenovo Chromebook). I often see similar issues in other program, too. I suppose this kind of issue is often not caught because English is one of the few languages without dead keys on their default keyboard...

**Comment by Jacalz** [comment]
Wow. Looking back at my old issues, this is actually working for me on the latest release. Closing :)
